.github/workflows/deploy.yml

```diff
@@ -0,0 +1,134 @@
+
+  pack:
+    name: Pack access-ram-env
+    runs-on: ubuntu-latest
+    needs: version
+    env:
+      VERSION: ${{ needs.version.outputs.version }}
+    outputs:
+      name: ${{ steps.access-ram.outputs.name }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get name
+        id: access-ram
+        run: echo "name=$(yq '.name' < env.yml)" >> $GITHUB_OUTPUT
+
+      - name: Setup Micromamba
+        uses: mamba-org/setup-micromamba@f8b8a1e23a26f60a44c853292711bacfd3eac822 #v1.9.0
+        with:
+          micromamba-version: '1.5.8-0'
+          environment-file: env.yml
+          environment-name: ${{ steps.access-ram.outputs.name }}-${{env.VERSION}}
+          generate-run-shell: true
+
+      - name: Create Pack and Lockfile
+        shell: micromamba-shell {0}
+        run: |
+          conda pack -o ${{ steps.access-ram.outputs.name }}-${{env.VERSION}}.tar.gz
+          conda-lock lock --file env.yml --platform linux-64 --micromamba --lockfile ${{ steps.access-ram.outputs.name }}-${{env.VERSION}}.conda-lock.yml
+
+      - name: Upload Artifact
+        env:
+          PACKAGE_NAME: ${{ steps.access-ram.outputs.name }}-${{ env.VERSION }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.PACKAGE_NAME }}
+          if-no-files-found: error
+          path: |
+            ${{ env.PACKAGE_NAME }}.tar.gz
+            ${{ env.PACKAGE_NAME }}.conda-lock.yml
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs:
+      - version
+      - pack
+      - setup-deployment-env
+    strategy:
+      fail-fast: false
+      matrix:
+        deployment-environment: ${{ fromJson(needs.setup-deployment-env.outputs.deployment-environments) }}
+    environment: ${{ matrix.deployment-environment }}
+    env:
+      VERSION: ${{ needs.version.outputs.version }}
+      PACKAGE_NAME: ${{ needs.pack.outputs.name }}-${{ needs.version.outputs.version }}
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          name: ${{ env.PACKAGE_NAME }}
+
+      - name: Set up SSH
+        uses: access-nri/actions/.github/actions/setup-ssh@main
+        id: ssh
+        with:
+          hosts: |
+            ${{ secrets.HOST_DATA }}
+            ${{ secrets.HOST }}
+          private-key: ${{ secrets.SSH_KEY }}
+
+      - name: Copy to ${{ matrix.deployment-environment }}
+        run: |
+          rsync -v -e 'ssh -i ${{ steps.ssh.outputs.private-key-path }}' \
+            ${{ env.PACKAGE_NAME }}.tar.gz \
+            ${{ secrets.USER }}@${{ secrets.HOST_DATA }}:${{ vars.PACK_DIR }}
+
+      - name: Deploy to ${{ matrix.deployment-environment }}
+        env:
+          DEPLOYMENT_DIR: ${{ vars.ENVIRONMENT_DIR }}/${{ env.VERSION }}
+        run: |
+          ssh ${{ secrets.USER }}@${{ secrets.HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<'EOT'
+          mkdir -p ${{ env.DEPLOYMENT_DIR }}
```

Comment:
`mkdir -p` will create parent directories if they don't exist which is good when the workflow runs for the first time. However if a directory already exists - if there's a pre-existing environment, it won't raise an error. So might be worth to check the contents of `${{ env.DEPLOYMENT_DIR }}` to be empty? Note: `tar -xzf ${{ vars.PACK_DIR }}/${{ env.PACKAGE_NAME }}.tar.gz -C ${{ env.DEPLOYMENT_DIR }}` requires `env.DEPLOYMENT_DIR` to exist but it's contents should be empty.

Though this may be redundant with the checks checking the version is updated in the PR?

Reply:
I think the the only way that would be possible is that someone pushes directly to the branch without a pull request, which would be an admin action...I feel like there would be enough safeguards to not need to check for this